openapi_python_client/parser/properties/model_property.py

```diff
@@ -20,6 +24,49 @@ class ModelProperty(Property):
 
     template: ClassVar[str] = "model_property.pyi"
 
+    def resolve_references(
+        self, components: Dict[str, Union[oai.Reference, oai.Schema]], schemas: Schemas
+    ) -> Union[Schemas, PropertyError]:
+        from ..properties import property_from_data
+
+        required_set = set()
+        props = {}
+        while self.references:
+            reference = self.references.pop()
+            source_name = Reference.from_ref(reference.ref).class_name
+            referenced_prop = components[source_name]
+            assert isinstance(referenced_prop, oai.Schema)
```

Comment:
Generally better to `cast` than to `assert` if just providing type info. Although there actually is no guarantee that this is true, is there? So this would really have to be a runtime check and return `PropertyError`